websocketOutportDriver/sender/webSocketSender_test.go

```diff
@@ -0,0 +1,213 @@
+package sender
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	coreMock "github.com/ElrondNetwork/elrond-go-core/core/mock"
+	"github.com/ElrondNetwork/elrond-go-core/websocketOutportDriver/data"
+	"github.com/ElrondNetwork/elrond-go-core/websocketOutportDriver/mock"
+	"github.com/gorilla/websocket"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewWebSocketSender(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil server", func(t *testing.T) {
+		t.Parallel()
+
+		args := getMockWebSocketSender()
+		args.Server = nil
+
+		wss, err := NewWebSocketSender(args)
+		require.Nil(t, wss)
+		require.Equal(t, data.ErrNilHttpServer, err)
+	})
+
+	t.Run("nil uint64 byte slice converter", func(t *testing.T) {
+		t.Parallel()
+
+		args := getMockWebSocketSender()
+		args.Uint64ByteSliceConverter = nil
+
+		wss, err := NewWebSocketSender(args)
+		require.Nil(t, wss)
+		require.Equal(t, data.ErrNilUint64ByteSliceConverter, err)
+	})
+
+	t.Run("nil logger", func(t *testing.T) {
+		t.Parallel()
+
+		args := getMockWebSocketSender()
+		args.Log = nil
+
+		wss, err := NewWebSocketSender(args)
+		require.Nil(t, wss)
+		require.Equal(t, data.ErrNilLogger, err)
+		require.True(t, wss.IsInterfaceNil())
+	})
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		args := getMockWebSocketSender()
+
+		wss, err := NewWebSocketSender(args)
+		require.NoError(t, err)
+		require.NotNil(t, wss)
+		require.False(t, wss.IsInterfaceNil())
+	})
+}
+
+func TestWebSocketSender_AddClient(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil client", func(t *testing.T) {
+		t.Parallel()
+
+		wss, _ := NewWebSocketSender(getMockWebSocketSender())
```

Comment:
```suggestion
		wss, err := NewWebSocketSender(getMockWebSocketSender())
		require.NoError(t, err)
```

Reply:
no need. I already have a test for the constructor